### PR TITLE
Rewrote fibonacci.py (#5665)

### PR DESCRIPTION
**Pull request by citharus**
### Describe your change:
I rewrote `fibonacci.py` as it was requested in #5665. The changes I made include: 
* Separated the logic to calculate Fibonacci numbers into the `get` method. 
* Removed `Out of Bound`, by calculating the missing numbers with the stored sequence. 
* Created `main` function with new prompt.
* Changed `get` method documentation.

* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters, with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] If this pull request resolves one or more open issues, then the commit message contains `Fixes: #{$ISSUE_NO}`.

Fixes: #5665

### CODE REVIEW

File: dynamic_programming/fibonacci.py [hunk 1]
```diff
@@ -5,61 +5,51 @@
...
-        elif N == 0:
-            self.fib_array.append(0)
-        print(self.fib_array)
+    def __init__(self) -> None:
+        self.sequence = [0, 1]
 
-    def get(self, sequence_no=None):
+    def calculate(self, index: int) -> None:
+        for _ in range(index):
+            self.sequence.append(self.sequence[-1] + self.sequence[-2])
```
Comment (poyea) [inline dynamic_programming/fibonacci.py:13]: Put inside `get`?

Comment (citharus) [reply dynamic_programming/fibonacci.py]: Yes makes sense

**Comment by tianyizheng02** [comment]
Thanks for the rewrite, @citharus! Make sure to link this PR to the original issue so GitHub knows to close the issue automatically when this PR gets merged (just `Fixes #5665` or something like that in your original comment will do)

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**